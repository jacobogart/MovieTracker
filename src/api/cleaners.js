import { imageUrl } from '../utils/API-logins';

export const cleanAllMovies = (moviesArray) => {
  const cleanMovies = moviesArray.map(movie => {
    const { 
      id, 
      title, 
      poster_path, 
      overview, 
      release_date, 
      vote_average 
    } = movie;

    return { 
      id, 
      title, 
      posterImg: `${imageUrl}${poster_path}`, 
      overview, 
      releaseDate: release_date, 
      rating: vote_average 
    };
  });

  return cleanMovies;
};

export const cleanPeople = (data) => {
  const director = data.crew.filter(person => person.job === "Director");
  const writer = data.crew.filter(person => person.job === "Writer");
  const cast = data.cast.map(person => {
    let { name, character, profile_path,  cast_id} = person;

    return { 
      name, 
      character, 
      headshot: `${imageUrl}${profile_path}`, 
      id: cast_id 
    };
  });

  return { director, writer, cast };
};