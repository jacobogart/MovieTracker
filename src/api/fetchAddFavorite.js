import { serverUrl } from "./pathNames";

export const fetchAddFavorite = (cardData) => {
  const url = `${serverUrl}/favorites/new`;
  const { 
    user, 
    title, 
    overview, 
    posterImg, 
    id, 
    rating, 
    releaseDate 
  } = cardData;
  let poster_path = posterImg.split('500')[1];
  const body = {
    movie_id: id, 
    user_id: user.id,
    title, 
    poster_path,
    release_date: releaseDate, 
    vote_average: rating, 
    overview
  };
  const init = {
    method: 'POST',
    body: JSON.stringify(body),
    headers: { 'Content-Type': 'application/json' }
  };

  return fetch(url, init)
    .then(response => {
      if (!response.ok) {
        throw Error('Failed to add favorite');
      } else {
        return response.json();
      }
    });
};
