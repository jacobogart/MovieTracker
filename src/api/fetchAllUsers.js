import { serverUrl } from "./pathNames";

export const fetchAllUsers = () => {
  return fetch(serverUrl)
    .then(response => {
      if (!response.ok) {
        throw new Error("Failed to create account");
      } else {
        return response.json();
      }
    })
    .then(data => data.data);
};
