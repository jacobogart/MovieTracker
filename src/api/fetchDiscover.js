import { jacobKey, baseUrl } from '../utils/API-logins'

const fetchDiscover = () => {
  return fetch(`${baseUrl}discover/movie${jacobKey}`)
    .then(response => {
      if (!response.ok) {
        throw new Error('Discover failed to fetch');
      } else {
        return response.json();
      }
  });
}

export default fetchDiscover;

