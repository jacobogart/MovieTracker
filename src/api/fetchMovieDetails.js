import { jacobKey } from '../utils/API-logins';
import { baseUrl } from './pathNames';

const fetchMoviePeople = (id) => {
  return fetch(`${baseUrl}movie/${id}/credits${jacobKey}`)
    .then(response => {
      if (!response.ok) {
        throw new Error('People failed to fetch');
      } else {
        return response.json();
      }
    });
};

export default fetchMoviePeople;
