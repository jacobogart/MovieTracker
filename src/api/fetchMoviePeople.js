import { jacobKey } from '../utils/API-logins';
import { baseUrl } from './pathNames';

const fetchMoviePeople = (id) => {
  return fetch(`${baseUrl}movie/${id}/credits${jacobKey}`)
    .then(response => {
      if (!response.ok) {
        throw new Error('Failed to fetch movie cast and crew');
      } else {
        return response.json();
      }
    });
};

export default fetchMoviePeople;
