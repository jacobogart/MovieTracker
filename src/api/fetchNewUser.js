import { serverUrl } from './pathNames';

export const fetchNewUser = (email, name, password) => {
  const body = JSON.stringify({ email, name, password });
  const init = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body
  };
  const request = new Request(`${serverUrl}/new`, init);

  return fetch(request)
    .then(response => {
      if (!response.ok) {
        throw new Error('Failed to create account');
      } else {
        return response.json();
      }
    })
    .then(data => {
      return data.id;
    });
};