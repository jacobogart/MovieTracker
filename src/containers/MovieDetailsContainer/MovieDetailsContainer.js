import React, { Component } from 'react';
import { connect } from 'react-redux';
import fetchMoviePeople from '../../api/fetchMovieDetails';
import { cleanPeople } from '../../api/cleaners';
import { MovieDetails } from '../../components/MovieDetails/MovieDetails';

class MovieDetailsContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      people: {},
      movieId: props.location.pathname.split("/")[2],
    }
  }


  componentDidMount() {
    const { movieId } = this.state;

    fetchMoviePeople(movieId)
    .then(data => cleanPeople(data))
    .then(people => this.setState({ people }));
  }
  
  render() {
    const movie = this.props.movies.find(movie => movie.id == this.state.movieId);
    const { people } = this.state;

    return (
      <article>
        {movie && people &&
          <MovieDetails 
            movieDetails={movie}
            people={people}
          />
        }
      </article>
    );
  }
}

export const mapStateToProps = (state) =>  ({
  movies: state.movies
});

export default connect(
  mapStateToProps,
  )(MovieDetailsContainer);