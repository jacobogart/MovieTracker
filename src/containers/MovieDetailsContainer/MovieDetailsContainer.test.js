import React from 'react';
import { shallow } from 'enzyme';
import { MovieDetailsContainer, mapStateToProps } from './MovieDetailsContainer';
import { cleanMovie, uncleanPeople, mockCleanPeople } from '../../api/__tests__/mockData';
import fetchMoviePeople from '../../api/fetchMoviePeople';
import { cleanPeople } from '../../api/cleaners';

const mockLocation = { pathname: '/localhost3001/11111'};
const mockMovies = [cleanMovie, cleanMovie]
jest.mock('../../api/fetchMoviePeople');
jest.mock("../../api/cleaners");

fetchMoviePeople.mockImplementation(() => Promise.resolve(uncleanPeople));
cleanPeople.mockImplementation(() => mockCleanPeople)

describe('MovieDetailsContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<MovieDetailsContainer location={mockLocation} movies={mockMovies} />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    const defaultState = {
      people: {},
      movieId: '11111'
    };
    wrapper = shallow(
      <MovieDetailsContainer
        location={mockLocation}
        movies={mockMovies}
      />,
      { disableLifecycleMethods: true });

    expect(wrapper.state()).toEqual(defaultState)
  });

  describe('componentDidMount', () => {
    it('should invoke fetchMoviePeople on mount', () => {
      expect(fetchMoviePeople).toHaveBeenCalledWith('11111');
    });

    it('should invoke cleanPeople', () => {
      expect(cleanPeople).toHaveBeenCalledWith(uncleanPeople);
    });

    it('should set state to cleaned people', () => {
      expect(wrapper.state('people')).toEqual(mockCleanPeople);
    });
  });

  describe('MSTP', () => {
    it('should return an object with a movies array', () => {
      const state = { movies: mockMovies, user: { name: 'Jacob' }}
      const expected = { movies: mockMovies };
      const result = mapStateToProps(state);
      expect(result).toEqual(expected);
    });
  });
});