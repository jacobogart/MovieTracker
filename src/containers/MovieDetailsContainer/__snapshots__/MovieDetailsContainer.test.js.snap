// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailsContainer should match snapshot 1`] = `<article />`;
